pkg/operator/apiserver/controller/apiservice/apiservice_controller.go

```diff
@@ -116,25 +100,29 @@ func (c *APIServiceController) sync() error {
 	}
 	ready, err := c.precondition(apiServices)
 	if err != nil {
-		v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+		if _, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
 			Type:    "APIServicesAvailable",
 			Status:  operatorv1.ConditionFalse,
 			Reason:  "ErrorCheckingPrecondition",
 			Message: err.Error(),
-		}))
+		})); updateErr != nil {
```

Comment:
for the reasons given in Slack, would keep returning `err`.

Reply:
Or at least combine both into one string.

Reply:
fixed.